service/history/workflow/mutable_state_impl.go

```diff
@@ -1843,6 +1866,27 @@ func (ms *MutableStateImpl) ApplyWorkflowExecutionStartedEvent(
 	ms.executionInfo.WorkflowExecutionTimeout = event.GetWorkflowExecutionTimeout()
 	ms.executionInfo.DefaultWorkflowTaskTimeout = event.GetWorkflowTaskTimeout()
 
+	ms.executionInfo.Callbacks = make(map[string]*persistencespb.CallbackInfo, len(event.GetCompletionCallbacks()))
+	for idx, cb := range event.GetCompletionCallbacks() {
+		// Use the start event version and history size as part of the callback ID to ensure that callbacks have
```

Comment:
what's `history size` here?

Reply:
I mean length, it's basically the ID of the start event.